### PR TITLE
Add config for per app maintenance windows for MHV on va.gov apps

### PR DESCRIPTION
**Pull request by stephenBarrs**
## Summary
Update maintenance windows configuration for MHV on va.gov apps - Medical Records, Medications and Secure Messaging.  This will allow each app to display a downtime banner independently if needed.  There is also a configuration that triggers all three apps to display downtime banner. 

| Pager Duty Service | vsp-infra-application-manifests | vets-website externalServices.js |
| ----------- | ----------- | ------------- |
| mhv-platform      | mhv_platform       | mhvPlatform: 'mhv_platform'  |
| mhv-medical-records   | mhv_mr        |  mhvMr: 'mhv_mr' |
| mhv-medications   | mhv_meds       | mhvMeds: 'mhv_meds'  |
| mhv-secure-messaging   | mhv_sm        | mhvSm: 'mhv_sm'  |

### Changes to Platform:
- add MHV app specific entries to externalServices

## Related issue(s)
- dependent on [this PR](https://github.com/department-of-veterans-affairs/vsp-infra-application-manifests/pull/2585) to merge the infra configuration in prod

## Testing done
- updated and added unit tests
- will also test in Staging 

## Screenshots
- Currently uses the default Downtime Notification banner so no change to UI

## Acceptance criteria
- [x] each app shows downtime banner when it should
- [x] each app does not show downtime banner when it should not

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Documentation has been updated ([link to documentation](https://github.com/department-of-veterans-affairs/va.gov-team/pull/72520/files#diff-db315a8adad6908236b3c8cd9f4d60ac6a40ac2dbdd57cd9048cdb50523450e5))
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication
- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

### CODE REVIEW

**Review by mmoyer-va — APPROVED** [review]
Looks good!

File: src/applications/mhv/medical-records/tests/containers/App.unit.spec.jsx [hunk 1]
```diff
@@ -66,7 +67,20 @@ describe('App', () => {
       dismissedDowntimeWarnings: [],
     },
   };
-
+  const downtime = maintenanceWindows => {
```
Comment (acrollet) [inline src/applications/mhv/medical-records/tests/containers/App.unit.spec.jsx:70]: probably not a good time with the move to the platform mhv dir, but maybe we could get out the DRYer and move this to a helper function later on :-)

File: src/applications/mhv/medical-records/containers/App.jsx [hunk 2]
```diff
@@ -184,7 +184,7 @@ const App = ({ children }) => {
         <MrBreadcrumbs />
         <DowntimeNotification
           appTitle="Medical Records"
-          dependencies={[externalServices.mhv]}
+          dependencies={[externalServices.mhvPlatform, externalServices.mhvMr]}
```
Comment (acrollet) [inline src/applications/mhv/medical-records/containers/App.jsx:187]: Just confirming if we want to depend on both services? Is it possible that the mhv platform service could be down without affecting already logged in users using the apps?

**Review by acrollet — APPROVED** [review]
Thanks for driving this forward @stephenBarrs ! I added a small comment and a question but this looks good :-)

**Review by alex-morgun-1 — APPROVED** [review]
Looks good from SM perspective

File: src/applications/mhv/medical-records/containers/App.jsx (hunk 2 above)
Comment (stephenBarrs) [reply src/applications/mhv/medical-records/containers/App.jsx:187]: The idea with mhvPlatform is that if it is set, then MR, SM and Meds will all show a Downtime Banner.  I did not rename the original `mhv` service in Pager Duty that is tied to the MHV Auth.

File: src/applications/mhv/medical-records/tests/containers/App.unit.spec.jsx (hunk 1 above)
Comment (stephenBarrs) [reply src/applications/mhv/medical-records/tests/containers/App.unit.spec.jsx:70]: I was thinking that we could even abstract the entire test for Maintenance Windows into a helper function that takes an array of service IDs and a boolean for if the Maintenance Window should be shown or not. If we move to custom Maintenance Banners we would have to rethink that, maybe pass in an array of assertions or something; future! It was pretty gut wrenching copy pasta-ing those very similar tests so many times!

File: src/applications/mhv/medical-records/containers/App.jsx (hunk 2 above)
Comment (acrollet) [reply src/applications/mhv/medical-records/containers/App.jsx:187]: oh of course, I forgot that you made a new service for mhv platform. ignore me, keep calm and carry on 😁

**Review by rsmithadhoc — APPROVED** [review]

**Pull request merged**